models/slice/matplotlib_slice_plotter.py

```diff
@@ -1,25 +1,26 @@
-from slice_plotter import SlicePlotter
-from mantid.simpleapi import AnalysisDataService
 from plotting import pyplot as plt
-
-
-def get_aspect_ratio(workspace):
-    return 'auto'
+from matplotlib.colors import Normalize, NoNorm
+from slice_plotter import SlicePlotter
 
 
 class MatplotlibSlicePlotter(SlicePlotter):
+    def __init__(self, slice_algorithm):
+        self._slice_algorithm = slice_algorithm
+        self._colormaps = ['viridis', 'jet', 'summer', 'winter']
+
     def display_slice(self, selected_workspace, x_axis, y_axis, smoothing, intensity_start, intensity_end, norm_to_one,
                       colourmap):
-        workspace = AnalysisDataService[selected_workspace]
-        ydata = []
-        for i in range(workspace.getNumberHistograms()-1,-1,-1):
-            ydata.append(workspace.readY(i))
-        x_left = workspace.readX(0)[0]
-        x_right = workspace.readX(0)[-1]
-        y_top = workspace.getNumberHistograms() - 1
-        y_bottom = 0
-        plt.imshow(ydata,extent=[x_left, x_right, y_bottom, y_top], aspect=get_aspect_ratio(workspace))
-
-
-
-
+            plot_data = self._slice_algorithm.compute_slice(selected_workspace, x_axis, y_axis, smoothing)
+            boundaries = [x_axis.start, x_axis.end, y_axis.start, y_axis.end]
+            if norm_to_one:
+                plot_data = self._slice_algorithm._norm_to_one(plot_data)
+            norm = Normalize(vmin=intensity_start, vmax=intensity_end)
+            plt.imshow(plot_data, extent=boundaries, interpolation='none', aspect='auto', cmap=colourmap, norm=norm)
```

Comment:
I'm not sure the model is the correct place to perform the display operations? It would seem to me that the we have the part to do with the the data, boundaries etc that belong to a model but the display should be performed by the presenter?